src/book/metadata.rs

```diff
@@ -1,12 +1,32 @@
+use std::path::PathBuf;
+
+/// TODO use in template: subtitle, description, publisher, number_format, section_names
+
 #[derive(Debug, Clone)]
 pub struct BookMetadata {
+    /// The title of the book.
     pub title: String,
+
+    /// TODO The subtitle, when titles are in the form of "The Immense Journey: An
+    /// Imaginative Naturalist Explores the Mysteries of Man and Nature"
+    pub subtitle: String,
+
+    /// TODO A brief description or summary.
     pub description: String,
 
+    /// TODO Publisher's info
+    pub publisher: Publisher,
+
     pub language: Language,
 
     authors: Vec<Author>,
     translators: Vec<Author>,
+
+    /// TODO Chapter numbering scheme
+    number_format: NumberFormat,
+    /// TODO Section names for nested Vec<Chapter> structures, such as `[
+    /// "Part", "Chapter", "Section" ]`
+    section_names: Vec<String>,
```

Comment:
This is probably a little too much anticipated. I am not against a high level of customisation, but I want to nail the basics first. 

I think we should concentrate on the fundamental necessities at first. Make the renderer abstraction work, make the plugin abstraction work. And after that we can add fine grained customisations. Trying to do everything at once will make it so much more difficult.